tests/main/econnreset/task.yaml

```diff
@@ -5,21 +5,25 @@ restore: |
 execute: |
     echo "Downloading a large snap in the background"
     su -c "/usr/bin/env SNAPD_DEBUG=1 snap download --edge test-snapd-huge 2>snap-download.log" test &
-    echo "Wait until it actually got some data for the snap"
-    while ! grep -q "Retrying.*download-snap/.*\.snap, attempt 1" snap-download.log; do
-        sleep 1
+
+    echo "Wait until the download started and downloaded more than 1 MB"
+    partial=test-snapd-huge_1.snap.partial
+    for i in $(seq 20); do
+        [ -f $partial ] && [ $(wc -c <"$partial") -gt 1048576 ] && break
```

Comment:
(nitpick) `wc -c < "$partial"` will actually open/read $partial to get the file size which can be expensive. A cheaper way to check for the size is something like `stat -c%s "$partial"`.